src/connectionManager.ts

```diff
@@ -0,0 +1,74 @@
+export interface Context {
+  id: string;
+  resourceID: string;
+  resourceType: string;
+  resourceDisplay?: string;
+}
+
+export interface Connection {
+  id: string;
+  url: URL;
+  context: Context[];
+}
+
+export class ConnectionManager {
+  private connections: Connection[];
+  private currentConnection?: Connection;
+
+  constructor() {
+    this.connections = [];
+  }
+
+  public getAllConnections(): Connection[] {
+    return this.connections;
+  }
+
+  public getCurrentConnection(): Connection | undefined {
+    return this.currentConnection;
+  }
+
+  public setCurrentConnection(connection: Connection | undefined) {
```

Comment:
Should this be a connection id? Passing in a whole connection here allows the current connection to be set to one that doesn't exist in the Manager in `this.connections`.